bazel/cc_toolchains/clang/toolchain_files.BUILD

```diff
@@ -110,6 +116,7 @@ filegroup(
         "as",
         "cov",
         "objcopy",
+        "nm",
```

Comment:
@JamesMBartlett @vihangm is it possible to get access to a toolchain's other tools? Ideally I'd use the toolchain version of `objdump` for the test added in #1976 (see discussion linked in this PR's description). Unlike `nm` and `objcopy`, which have [custom Makefile variables](https://bazel.build/reference/be/make-variables#custom_variables) built-in, `objdump` doesn't have an existing variable.

It seems that [TemplateVariableInfo](https://www.stevenengelhardt.com/2023/03/03/practical-bazel-custom-bazel-make-variables/) could work for this, but it feels a little strange that it wouldn't be toolchain specific the way `@bazel_tools//tools/cpp:current_cc_toolchain` is.